File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        // 최대 2명만 태울수 있다.

        Arrays.sort(people);  // 오름차순 정렬
        Deque<Integer> dq = new LinkedList<>();
        for (int num : people) {
            dq.add(num);
        }
        while(dq.size()>=2){
            int left=dq.peek();
            int right=dq.getLast();
            if(left+right <= limit){
                dq.removeFirst();
            }
            dq.removeLast();
            answer++;
        }
        if(!dq.isEmpty()){
            answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int solution(vector<int> people, int limit) {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int answer = 0;

    sort(people.begin(), people.end());
    int start = 0;
    int end = static_cast<int>(people.size()-1);

    while (start <= end)
    {
        if (people.at(start) + people.at(end) <= limit)
        {
            start++;
        }
        end--;
        answer++;
    }
    cout << answer << "\n";

    return answer;
}